rusk/src/lib/node/rusk.rs

```diff
@@ -737,11 +738,12 @@ fn execute(
 
     // Deploy if this is a deployment transaction and spend part is successful.
     if let Some(deploy) = tx.deploy() {
+        let gas_left = tx.gas_limit() - receipt.gas_spent;
         if receipt.data.is_ok() {
             contract_deploy(
                 session,
                 deploy,
-                tx.gas_limit(),
+                gas_left,
```

Comment:
passing here the decreasing gas limit will cause double subtraction down below, this is a good change in general but we need to prevent double subtraction